chevah/compat/tests/normal/testing/test_testcase.py

```diff
@@ -380,6 +380,42 @@ def much_later():  # pragma: no cover
         delayed_call.cancel()
         self.executeReactor()
 
+    def test_cleanReactor_delayed_calls_all_active(self):
+        """
+        It  will cancel any delayed calls in the reactor queue.
+        """
+        reactor.callLater(1, self.ignoreFailure)
+        reactor.callLater(2, self.ignoreFailure)
+        self.assertIsNotEmpty(reactor.getDelayedCalls())
+
+        self._cleanReactor()
+
+        self.assertIsEmpty(reactor.getDelayedCalls())
+
+    def test_cleanReactor_delayed_calls_some_called(self):
+        """
+        It  will not break if a call is already called and will continue
+        canceling the .
+        """
+        delayed_call_1 = reactor.callLater(1, self.ignoreFailure)
+        delayed_call_2 = reactor.callLater(2, self.ignoreFailure)
+        # Fake that deferred was called and make sure it is first in the list
+        # so that we can can check that the operation will continue.
+        delayed_call_1.called = True
```

Comment:
> o when we use clock, there is no need to clean it...

Agreed, I've played with the test and the Clock and that was my design as well.

> but the code here is for functional tests where we are using the real reactor, and in which we are re-using the same reactor ... and accessing the same reactor from a thread so that we can have the service in one thread and the testing bot in another thread

I know, just that functional and mocking/hacking don't go well together and I was looking for a better way.